pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,69 @@
+package backuphelpers
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+)
+
+type Enqueueable interface {
+	Enqueue()
+}
+
+type BackupVar interface {
+	AddListener(listener Enqueueable)
+	SetBackupSpec(spec backupv1alpha1.EtcdBackupSpec)
+	ArgString() string
+}
+
+type BackupConfig struct {
+	enabled   bool
+	spec      backupv1alpha1.EtcdBackupSpec
+	listeners []Enqueueable
+	mux       sync.Mutex
+}
+
+func NewDisabledBackupConfig() *BackupConfig {
+	return &BackupConfig{
+		enabled: false,
+		mux:     sync.Mutex{},
+	}
+}
+
+func (b *BackupConfig) SetBackupSpec(spec backupv1alpha1.EtcdBackupSpec) {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	b.spec = spec
+
+	for _, l := range b.listeners {
+		l.Enqueue()
+	}
```

Comment:
deadlock, why ?

only one controller can hold the lock, no ?

Reply:
if you notify the other controller it will call the reading argument function, which can't acquire the lock

Reply:
and you're also introducing the exact same issue I've been fixing in
https://github.com/openshift/cluster-etcd-operator/pull/1300

we need to check whether the spec has changed, otherwise the other controller will hotloop around every invocation of SetBackupSpec

Reply:
ahhh .. so the function wont return till it holds the lock 

but this controller will eventually release the lock, no ?

Reply:
depends how it's implemented on the other side, but let's leave this if the CI is OK